### PR TITLE
3481 if there is more than 20 failed tests add another notification to saleor-qa channel

### PR DESCRIPTION
**Pull request by karola312**
I want to merge this change because we want a notification on slack if more than 20 tests failed or errored. This slack notifications should appear after nightly tests, or on workflow dispatch

**PR intended to be tested with API branch:** 

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants
23. [ ] payments

CONTAINERS=1

### CODE REVIEW

File: .github/workflows/getFailedTests.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+  const throwErrorAfterTimeout = setTimeout(function () {
+    throw new Error("Run have still running status, after all tests executed");
+  }, 1200000);
+
+  const data = await waitForTestsToFinish(requestVariables);
+
+  clearTimeout(throwErrorAfterTimeout);
+  return { status: data.status, runId: data.id };
+}
```
Comment (NyanKiyoshi) [inline .github/workflows/getFailedTests.js:45]: We already have such logic, could you create a module and reuse it instead? 
- https://github.com/saleor/saleor-dashboard/blob/607b5d40a5a69b2be11e3d6418cf16694fd6a077/.github/workflows/approveAndMergeReleasePR.js#L179-L204
- https://github.com/saleor/saleor-dashboard/blob/607b5d40a5a69b2be11e3d6418cf16694fd6a077/.github/workflows/approveAndMergeReleasePR.js#L136-L153

File: .github/workflows/tests-nightly.yml [hunk 2]
```diff
@@ -315,5 +317,43 @@ jobs:
...
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+
+      - name: Install dependencies
+        run: |
+              cd .github/workflows
+              npm ci
```
Comment (mwidera) [inline .github/workflows/tests-nightly.yml:339]: You can use builtin directory indicator here:
```suggestion
      - name: Install dependencies
        working-directory: .github/workflows
        run: npm ci
```

Also: whole job `send-slack-notification-scheduled-and-manually` runs inside the `.github/workflows`, thus maybe you could define `default` for that?
Like this: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-set-the-default-shell-and-working-directory

Comment (mwidera) [reply .github/workflows/tests-nightly.yml]: Thanks!

**Review by mwidera — DISMISSED** [review]

**Review by mwidera — APPROVED** [review]

File: .github/workflows/tests-nightly.yml [hunk 3]
```diff
@@ -308,12 +310,56 @@ jobs:
...
+            node cypressTestsHelpers/getFailedTests.js \
+            --dashboard_url $dashboard_url 
+
+      - name: Notify Slack
+        if: steps.get-tests-results.outputs.testStatus == 'true'
+        env:
+          JOB_DEPLOYMENT_KIND: ${{ needs.run-tests-in-parallel.outputs.environment }}
+          JOB_STATUS: ${{ job.status }}
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_SALEOR_QA_WEBHOOK_URL }}
+          JOB_TITLE: "A lot of tests failed, something is probably broken - ${{ needs.run-tests-in-parallel.outputs.dashboard_url }}"
```
Comment (wojteknowacki) [inline .github/workflows/tests-nightly.yml:362]: Maybe we could add here also ENV_URI

Comment (wojteknowacki) [reply .github/workflows/tests-nightly.yml:362]: And link to the dashboard itself where it occurred.

Comment (karola312) [reply .github/workflows/tests-nightly.yml:362]: In title we will have cypress dashboard url - needs.run-tests-in-parallel.outputs.dashboard_url, in text there will be something like "GithubBot: Tests finished for HTTPS://AUTOMATION-DASHBOARD.STAGING.SALEOR.CLOUD/, result: failure"

**Review by szczecha — APPROVED** [review]

**Pull request merged**